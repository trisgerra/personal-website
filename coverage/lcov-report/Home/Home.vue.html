<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for Home/Home.vue</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../index.html">All files</a> / <a href="index.html">Home</a> Home.vue
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>7/7</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>6/6</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">10x</span>
<span class="cline-any cline-yes">10x</span>
<span class="cline-any cline-yes">10x</span>
<span class="cline-any cline-yes">10x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">10x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">&lt;template&gt;
  &lt;div class="home"&gt;
    &lt;head-hero&gt;&lt;/head-hero&gt;
    &lt;div class="card-container"&gt;
      &lt;div class="columns"&gt;
        &lt;div v-for="card in cards" v-bind:key="card.cardName" class="column home-card"&gt;
          &lt;home-card :card="card"&gt;&lt;/home-card&gt;
        &lt;/div&gt;
      &lt;/div&gt;
    &lt;/div&gt;
  &lt;/div&gt;
&lt;/template&gt;
&nbsp;
&lt;script lang="ts"&gt;
import { Component, Vue } from 'vue-property-decorator';
import HeadHero from './HeadHero/HeadHero.vue';
import HomeCard from './HomeCard/HomeCard.vue';
import { homeCards, ICard } from './homeCardItems';
&nbsp;
@Component({
  components: {
    HeadHero,
    HomeCard,
  },
})
export default class Home extends Vue {
  cards: ICard[] = homeCards;
}
&lt;/script&gt;
&nbsp;
&lt;style lang="scss" scoped&gt;
.columns {
    margin: 0px;
    width: 100%;
}
.column {
    padding:0px;
}
.card-container {
  border-top: 3px solid #FF0000;
  border-right: 3px solid #FF0000;
}
&lt;/style&gt;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sat Jul 13 2019 00:06:37 GMT+0200 (Central European Summer Time)
</div>
</div>
<script src="../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../sorter.js"></script>
</body>
</html>
