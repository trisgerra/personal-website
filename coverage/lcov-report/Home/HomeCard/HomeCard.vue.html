<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for Home/HomeCard/HomeCard.vue</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">Home/HomeCard</a> HomeCard.vue
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>6/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>5/5</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">10x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">10x</span>
<span class="cline-any cline-yes">10x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">9x</span>
<span class="cline-any cline-yes">9x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">&lt;template&gt;
  &lt;div :class="cardClass"&gt;
    &lt;div class="card-content"&gt;
      &lt;h3 class="card-title"&gt;{{card.name}}&lt;/h3&gt;
      &lt;p v-for="item in card.items"
      v-bind:key="item.text"
      class="item"
      v-bind:style="{'border-left': '3px solid ' + item.color}"&gt;{{item.text}}&lt;/p&gt;
    &lt;/div&gt;
  &lt;/div&gt;
&lt;/template&gt;
&nbsp;
&lt;script lang="ts"&gt;
import { Component, Vue, Prop } from 'vue-property-decorator';
import { ICard } from '../homeCardItems';
@Component({
  components: {},
})
export default class HomeCard extends Vue {
  @Prop() card!: ICard;
&nbsp;
  get cardClass(): string {
    const { isDark } = this.card;
    return isDark ? 'card column dark' : 'card column';
  }
}
&nbsp;
&lt;/script&gt;
&nbsp;
&lt;style lang="scss" scoped&gt;
.card {
  height: 100%;
  color: #151616;
}
&nbsp;
.card-content {
  padding-top: 60px;
  padding-bottom: 60px;
  padding-left: 60px;
}
&nbsp;
.dark {
  color: white;
  background-color: #151616;
}
h3 {
  margin-bottom: 16px;
  font-size: 32px;
  font-weight: bold;
}
p {
  margin: 0px 30px 8px 3px;
  padding: 0px 8px 0px 8px;
  border-left: 3px solid;
}
&lt;/style&gt;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sat Jul 13 2019 00:06:37 GMT+0200 (Central European Summer Time)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
