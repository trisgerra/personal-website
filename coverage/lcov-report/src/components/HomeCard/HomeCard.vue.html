<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/HomeCard/HomeCard.vue</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">src/components/HomeCard</a> HomeCard.vue
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>0/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>0/5</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">&lt;template&gt;
  &lt;div :class="cardClass"&gt;
    &lt;div class="card-content"&gt;
      &lt;h3 class="card-title"&gt;{{card.name}}&lt;/h3&gt;
      &lt;p v-for="item in card.items"
      v-bind:key="item.text"
      class="item"
      v-html="item.text"
      v-bind:style="{'border-left': '3px solid ' + item.color}"&gt;&lt;/p&gt;
    &lt;/div&gt;
  &lt;/div&gt;
&lt;/template&gt;
&nbsp;
&lt;script lang="ts"&gt;
<span class="cstat-no" title="statement not covered" >import { Component, Vue, Prop } from 'vue-property-decorator';</span>
import { ICard } from '../homeCardItems';
@Component({
  components: {},
})
<span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" >export default class HomeCard extends Vue {</span></span>
<span class="cstat-no" title="statement not covered" >  @Prop() card!: ICard;</span>
&nbsp;
<span class="fstat-no" title="function not covered" >  get cardClass(): string {</span>
<span class="cstat-no" title="statement not covered" >    const { isDark } = this.card;</span>
<span class="cstat-no" title="statement not covered" >    return isDark ? 'card column dark' : 'card column';</span>
  }
}
&nbsp;
&lt;/script&gt;
&nbsp;
&lt;style lang="scss" scoped&gt;
.card {
  height: 100%;
  color: #151616;
}
&nbsp;
.card-content {
  padding-top: 60px;
  padding-bottom: 60px;
  padding-left: 60px;
}
&nbsp;
.dark {
  color: white;
  background-color: #151616;
}
h3 {
  margin-bottom: 16px;
  font-size: 32px;
  font-weight: bold;
}
p {
  margin: 0px 30px 8px 3px;
  padding: 0px 8px 0px 8px;
  border-left: 3px solid;
}
&lt;/style&gt;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sat Apr 11 2020 18:41:15 GMT+0200 (Central European Summer Time)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
