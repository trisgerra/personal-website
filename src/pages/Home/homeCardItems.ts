interface IItem {
    text: string;
    color: string;
}

export interface ICard {
    name: string;
    items: IItem[],
    isDark: boolean
}

export interface ISquareItem {
  title: string;
  color: string;
  url: string;
}

const experienceList : IItem[] = [
  {
    text: '<b>2012</b> - iOS Internship',
    color: '#FF0000',
  },
  {
    text: '<b>2013</b> - iOS Developer',
    color: '#FF0000',
  },
  {
    text: '<b>2014</b> - Mobile Developer',
    color: '#FF0000',
  },
  {
    text: '<b>2015</b> - Mobile Developer & Analyst',
    color: '#FF0000',
  },
  {
    text: '<b>2016</b> - Mobile Developer, Analyst Designer',
    color: '#FF0000',
  },
  {
    text: '<b>2017</b> - Mobile Lead',
    color: '#FF0000',
  },
  {
    text: '<b>2018</b> - Frontend Developer',
    color: '#00D1FF',
  },
  {
    text: '<b>Now</b> - Senior Frontend Developer',
    color: '#00D1FF',
  },
];

const skillsList : IItem[] = [
  {
    text: '<b>iOS</b> - [Obj-C, Swift, React Native]',
    color: '#FF0000',
  },
  {
    text: '<b>Android</b> - [Java, Swift, React Native]',
    color: '#FF0000',
  },
  {
    text: '<b>Web</b> - [Vue, React]',
    color: '#00D1FF',
  },
  {
    text: '<b>Unit Testing</b> - [Jest]',
    color: '#00D1FF',
  },
  {
    text: '<b>E2E Testing</b> - [Cypress]',
    color: '#00D1FF',
  },
  {
    text: '<b>Design</b> - [Invision Studio, Sketch, Photoshop]',
    color: '#151616',
  },
  {
    text: '<b>CI/CD</b> - [Travis CI, Circle CI, CodeClimate, Jenkins, Heroku]',
    color: '#151616',
  },
  {
    text: '<b>Devops</b> - [Docker, Kubernetes]',
    color: '#151616',
  },
];

const hobbiesList : IItem[] = [
  {
    text: '<b>Vaping</b>',
    color: '#FF0000',
  },
  {
    text: '<b>Drive simulator</b>',
    color: '#FF0000',
  },
  {
    text: '<b>Fitness</b>',
    color: '#00D1FF',
  },
  {
    text: '<b>Politics</b>',
    color: '#00D1FF',
  },
  {
    text: '<b>Cryptocurrencies</b>',
    color: '#00D1FF',
  },
];

export const homeCards : ICard[] = [
  {
    name: 'WORK EXPERIENCE',
    items: experienceList,
    isDark: true,
  },
  {
    name: 'SKILLS',
    items: skillsList,
    isDark: false,
  },
  {
    name: 'HOBBIES',
    items: hobbiesList,
    isDark: true,
  },
];

export const squareCards : ISquareItem[][] = [
  [{
    title: 'GITHUB',
    color: '#151616',
    url: 'https://github.com/trisgerra',
  },
  {
    title: 'LINKEDIN',
    color: '#2D2F2F',
    url: 'https://www.linkedin.com/in/a-digiacomo/',
  },
  {
    title: 'TWITCH',
    color: '#494A4A',
    url: 'https://www.twitch.tv/asraces',
  },
  {
    title: 'CONTACT',
    color: '#2D2F2F',
    url: 'mailto:andrea.digiacomo@live.it',
  }],
];
