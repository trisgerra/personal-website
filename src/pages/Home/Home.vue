<template>
  <div class="home">
    <head-hero></head-hero>
    <div class="card-container">
      <div class="columns">
        <div v-for="card in cards" v-bind:key="card.cardName" class="column home-card">
          <home-card :card="card"></home-card>
        </div>
      </div>
    </div>
    <div class="square-container">
      <div v-bind:key="index" v-for="(row, index) in squares" class="square-row">
        <div class="columns">
          <div v-for="item in row" v-bind:key="item.title" class="column square-card">
            <square-card :item="item"></square-card>
          </div>
        </div>
      </div>
    </div>
  </div>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator';
import HeadHero from './HeadHero/HeadHero.vue';
import HomeCard from './HomeCard/HomeCard.vue';
import SquareCard from './SquareCard/SquareCard.vue';
import {
  homeCards, squareCards, ICard, ISquareItem,
} from './homeCardItems';

@Component({
  components: {
    HeadHero,
    HomeCard,
    SquareCard,
  },
})
export default class Home extends Vue {
  cards: ICard[] = homeCards;

  squares: ISquareItem[][] = squareCards;
}
</script>

<style lang="scss" scoped>
.columns {
  margin: 0px;
  width: 100%;
}
.column {
  padding: 0px;
}
.card-container {
  border-top: 3px solid #ff0000;
  border-right: 3px solid #ff0000;
}
.square-container {
    border-right: 3px solid #ff0000;
}
</style>
