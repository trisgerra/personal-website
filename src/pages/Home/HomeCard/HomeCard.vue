<template>
  <div :class="cardClass">
    <div class="card-content">
      <h3 class="card-title">{{card.name}}</h3>
      <p v-for="item in card.items"
      v-bind:key="item.text"
      class="item"
      v-bind:style="{'border-left': '3px solid ' + item.color}">{{item.text}}</p>
    </div>
  </div>
</template>

<script lang="ts">
import { Component, Vue, Prop } from 'vue-property-decorator';
import { ICard } from '../homeCardItems';
@Component({
  components: {},
})
export default class HomeCard extends Vue {
  @Prop() card!: ICard;

  get cardClass(): string {
    const { isDark } = this.card;
    return isDark ? 'card column dark' : 'card column';
  }
}

</script>

<style lang="scss" scoped>
.card {
  height: 100%;
  color: #151616;
}

.card-content {
  padding-top: 60px;
  padding-bottom: 60px;
  padding-left: 60px;
}

.dark {
  color: white;
  background-color: #151616;
}
h3 {
  margin-bottom: 16px;
  font-size: 32px;
  font-weight: bold;
}
p {
  margin: 0px 30px 8px 3px;
  padding: 0px 8px 0px 8px;
  border-left: 3px solid;
}
</style>
