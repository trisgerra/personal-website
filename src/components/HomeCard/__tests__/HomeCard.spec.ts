import { Wrapper } from '@vue/test-utils';
import HomeCard from '../HomeCard.vue';
import createWrapper from '@/utils/CreateWrapper';
import { homeCards } from '../../../pages/Home/homeCardItems';

describe('HomeCard', () => {
  const wrapper: Wrapper<HomeCard> = createWrapper(HomeCard, {
    propsData: {
      card: homeCards[0],
    },
  });

  it('should render as expected', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('wrapper is a card', () => {
    expect(wrapper.is('.card')).toBeTruthy();
  });

  it('contains the correct title', () => {
    expect(wrapper.findAll('.card-title').length).toBe(1);
    expect(wrapper.find('.card-title').text()).toEqual('WORK EXPERIENCE');
  });

  it('contains the correct number of item', () => {
    expect(wrapper.findAll('.item').length).toBe(8);
    expect(wrapper.find('.item').text()).toEqual('2012 - iOS Internship');
  });
});
