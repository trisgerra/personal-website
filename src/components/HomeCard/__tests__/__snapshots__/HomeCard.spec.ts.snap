// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeCard should render as expected 1`] = `
<div
  class="card column dark"
>
  <div
    class="card-content"
  >
    <h3
      class="card-title"
    >
      WORK EXPERIENCE
    </h3>
     
    <p
      class="item"
      style="border-left: 3px solid #FF0000;"
    >
      <b>
        2012
      </b>
       - iOS Internship
    </p>
    <p
      class="item"
      style="border-left: 3px solid #FF0000;"
    >
      <b>
        2013
      </b>
       - iOS Developer
    </p>
    <p
      class="item"
      style="border-left: 3px solid #FF0000;"
    >
      <b>
        2014
      </b>
       - Mobile Developer
    </p>
    <p
      class="item"
      style="border-left: 3px solid #FF0000;"
    >
      <b>
        2015
      </b>
       - Mobile Developer & Analyst
    </p>
    <p
      class="item"
      style="border-left: 3px solid #FF0000;"
    >
      <b>
        2016
      </b>
       - Mobile Developer, Analyst Designer
    </p>
    <p
      class="item"
      style="border-left: 3px solid #FF0000;"
    >
      <b>
        2017
      </b>
       - Mobile Lead
    </p>
    <p
      class="item"
      style="border-left: 3px solid #00D1FF;"
    >
      <b>
        2018
      </b>
       - Frontend Developer
    </p>
    <p
      class="item"
      style="border-left: 3px solid #00D1FF;"
    >
      <b>
        Now
      </b>
       - Senior Frontend Developer
    </p>
  </div>
</div>
`;
